From a88db85b735c4e9ba7732210c1ed534035e00c81 Mon Sep 17 00:00:00 2001
From: Lin Xie <lin.xie@intel.com>
Date: Fri, 22 Nov 2019 13:29:14 +0800
Subject: [PATCH] Add empty filter graph check for profiling

---
 fftools/ffmpeg.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/fftools/ffmpeg.c b/fftools/ffmpeg.c
index 7a2863d..771cd83 100644
--- a/fftools/ffmpeg.c
+++ b/fftools/ffmpeg.c
@@ -1685,8 +1685,9 @@ static void print_report(int is_last_report, int64_t timer_start, int64_t cur_ti
     if (init_time == 0 && do_profiling_all) {
         for (i = 0; i < nb_filtergraphs; i++) {
             FilterGraph *fg = filtergraphs[i];
-            int j;
-            for (j = 0; j < fg->graph->nb_filters; j++) {
+            if (!fg || !fg->graph)
+                continue;
+            for (int j = 0; j < fg->graph->nb_filters; j++) {
                 AVFilterContext *ft = fg->graph->filters[j];
                 if (!ft)
                     continue;
@@ -4837,8 +4838,9 @@ static int transcode(void)
     if (do_profiling_all) { //for filters
         for (i = 0; i < nb_filtergraphs; i++) {
             FilterGraph *fg = filtergraphs[i];
-            int j;
-            for (j = 0; j < fg->graph->nb_filters; j++) {
+            if (!fg || !fg->graph)
+                continue;
+            for (int j = 0; j < fg->graph->nb_filters; j++) {
                 AVFilterContext *ft = fg->graph->filters[j];
                 int64_t frame_cnt = 0;
                 int k;
-- 
2.7.4

